### PR TITLE
Change "categories" to "tags" in front matter and make blog permalinks shorter

### PR DESCRIPTION
**Pull request by cwang395**
Setting categories in the frontmatter is equivalent to putting the blog posts in folders, which was why the URLs to individual posts were very long. This change replaces "categories" with "tags" in the frontmatter of the blog posts, which was probably what the authors were going for. But it **will** break existing links to the blog posts, so I'm not sure whether to merge it or not. 😰

### CODE REVIEW

**Comment by jiangdaniel** [comment]
Since there are only six blog posts right now, we could add redirects for all of them either in JavaScript or on Cloudflare.

**Comment by cwang395** [comment]
Awesome! I don't know the CloudFlare credentials but for the record here are the redirects that would need to happen:

| Original URL | New URL |
|--------------|----------|
| /2016,/scholarship/2016/05/01/scholarship-winners.html | /blog/2016/scholarship-winners |
| /google/2015/11/09/Google-Monitors.html | /blog/2015/google-monitors |
| /annual/report/final/competition/scholarship/2015/10/31/annual-report.html | /blog/2015/annual-report |
| /alumni/application/kickoff/scholarship/workshop/2015/06/06/Scholarship.html | /blog/2015/scholarship |
| /alumni/application/kickoff/scholarship/workshop/2015/04/28/competition-awards.html | /blog/2015/competition-awards |
| /sponsorship/cctv/livestream/cameras/2015/03/17/new-livestreaming-systems.html | /blog/2015/livestreaming-systems |

Basically, the format is /blog/:year/:title, where :year is the year the post was written and :title is the blog post filename after the date part. I lowercased some of the blog post filenames to keep things consistent.

**Comment by jiangdaniel** [comment]
Actually, since we're trying to fix historical blog post links, there will be quite a few links, right?  It might be better to use JavaScript on the 404 page.  What do you think about using JavaScript to parse everything out of the old urls except for blog/:year/:title?

**Comment by cwang395** [comment]
What do you mean by historical blog post links? So far, we only have these 6 links on our blog. Historical blog posts will have to be copied into the `_posts/` folder manually anyways, and the front matter added manually to each post. At that point, we can just name the files and add the right front matter.

**Comment by scottnuma** [comment]
In my pull request #240, I added in all the old blog posts with front matter. Unfortunately, I didn't see your pull request and parsed them in using tags.'

Hmm, this seems a little complicated - maybe we should talk about this Wednesday?

**Comment by cwang395** [comment]
So here's the plan:

1. Make the urls for all blog posts `/blog/year/month/day/title` format (not `/blog/year/title`) in `_config.yml`, to stay consistent with our blog urls on the old site.
1. For the 6 existing blog posts on the new site that have crazy long urls, create dummy files with the old filenames and the old URLs in the `permalink` field of the front matter. Add a `redirect_to` field that will redirect them to the new, `/blog/year/month/day/title` formatted url. This lets us clearly document what the url mappings are. ([source](https://superdevresources.com/redirects-jekyll-github-pages/))
2. Merge the blog posts in #240; they should have the same urls as the blog posts on the old site.

I'll update this pull request when I'm done.

**Comment by cwang395** [comment]
See staged changes at [cwang395.github.io/website/blog/](http://cwang395.github.io/website/blog/).
